server/base/src/main/java/org/apache/accumulo/server/util/RestoreZookeeper.java

```diff
@@ -118,6 +118,9 @@ public static void main(String[] args) throws Exception {
     }
 
     SAXParserFactory factory = SAXParserFactory.newInstance();
+    factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+    factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+    factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
```

Comment:
These features are very specific to a particular parser. The only parser that we can reasonably expect any users to be using is xerces-j 2.7.1, which was integrated into `rt.jar` since JDK6.

So, I would just do the following instead of anything more complicated or speculative with respect to a particular parser:
`factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);`

If the user is really convoluting their environment to use a different parser, then I think they're sort of on their own anyway. But, this tool is typically run in the same security context as the user... and not by a remote service, so there really isn't a vulnerability out-of-the-box anyway.